Vapor 5

0xTim opened the pull request:
![Vapor5](https://github.com/user-attachments/assets/a77e0bcb-5210-43d1-a131-38c58bbb524a)

Jinxiansen commented:
Awesome logo~

lmcd commented:
Write up here: https://blog.vapor.codes/posts/the-future-of-vapor/

AltiAntonov commented on Sources/Vapor/Passwords/Application+Password.swift:
```diff
@@ -1,40 +1,40 @@
-        }
-    }
-}
+//import Foundation
```
Maybe add #warning("TODO") to not forget removing the file.

AltiAntonov commented on Sources/Vapor/Passwords/Application+Passwords.swift:
```diff
@@ -1,53 +1,53 @@
-        }
-    }
-}
+//import NIOConcurrencyHelpers
```
Maybe add #warning("TODO") to not forget removing the file.

0xTim committed:
Remove noasync command stuff

0xTim committed:
More removals and renames

0xTim committed:
Make handlers async

0xTim committed:
Make server protocol async

0xTim committed:
Remove noasync APIs from HTTPServer

0xTim committed:
Default to port 0

0xTim committed:
Migrate SessionDriver to Async

0xTim committed:
Migrate SessionTests to Swift Testing

0xTim committed:
Migrate CredentialsAuthenticator to async

0xTim committed:
Migrate auth tests to Swift Testing

0xTim committed:
Start migrating authenticator to async

0xTim committed:
More auth migrations

0xTim committed:
Final Async Auth tidy up

0xTim committed:
Migrate Cache to concurrency

0xTim committed:
Migrate ViewRenderer to Concurrency

0xTim committed:
Remove NonBlockingFileIO

0xTim committed:
Remove futures from View

0xTim committed:
Migrate password tests to Swift Testing

0xTim committed:
Migrate Client to async

0xTim committed:
Migrate endopint cache to async

0xTim committed:
Fix tests

0xTim committed:
Make sure we only make one endpoint cache request at once

0xTim committed:
Start removing futures

0xTim committed:
Migrate CORSMiddleware

0xTim committed:
Merge AsyncRequestDecodable into RequestDecodable

0xTim committed:
Migrate stuff over to AsyncResponseEncodable

0xTim committed:
Start removing some futures

0xTim committed:
Add service lifecycle

0xTim committed:
Add Server from Hummingbird

0xTim committed:
Start migrating Pipeline tests to use an async testing channel and Swift Testing

0xTim committed:
Migrate tests that fail with async middleware to use a async testing channel

0xTim committed:
Migrate the Pipeline tests to Swift Testing and NIOAsyncTestChannel

0xTim committed:
Remove some futures

0xTim committed:
Start migrating to new server

0xTim committed:
Remove oncloseFuture now we're async

0xTim committed:
Align the packages

0xTim committed:
Hack in new server

0xTim committed:
Merge branch 'main' into vapor-5

0xTim committed:
Rename

0xTim commented:
Marking ready for review to trigger CI

0xTim committed:
Migrate Bcrypt tests to Swift testing

0xTim committed:
Migrate more tests to Swift Testing

0xTim committed:
Test migration

0xTim commented:
In fact I'm going to mark this as ready for review so I can get it running through CI and catch issues early as it's gonna be a big one

0xTim committed:
Get it compiling on Swift 6

0xTim committed:
Merge branch 'main' into vapor-5

0xTim committed:
Resolve some test issues

0xTim committed:
Fix race in test

0xTim committed:
Migrate Application Tests

0xTim committed:
Migrate DotEnv tests

0xTim committed:
Migrate environment and base tests

0xTim committed:
Migrate BodyStreamStateTests

0xTim committed:
Migrate MetricsTests

0xTim committed:
Add new test target for tests that need to be run sequentially

0xTim committed:
Migrate the File Tests

0xTim committed:
Migrate ErrorTests

0xTim committed:
Fix usage of XCTVapor

0xTim committed:
Migrate websocket tests

0xTim committed:
Migrate OTP Tests

0xTim committed:
Migrate response compression tests

0xTim committed:
Migrate HTTPMediaTypeSetTests

0xTim committed:
Migrae content tests

0xTim committed:
Remove some force trys

0xTim committed:
Do it the 'proper' way

0xTim committed:
Migrate HTTPCacheTests

0xTim committed:
Convert URLEncodedFormTests

0xTim committed:
Inheritance is the devil

0xTim committed:
Migrate URITests

0xTim committed:
Migrate the service tests

0xTim committed:
Migrate header tests

0xTim committed:
Fix some warnings

0xTim committed:
More warnings

0xTim committed:
Try test traits for metrics

0xTim committed:
Attempt to wrap the metrics factory so it ferries the call down to the right task local

0xTim committed:
Simplify stuff

0xTim committed:
Migrate the valiation tests

0xTim committed:
Migrate more tests

0xTim committed:
Migrate Route Tests

0xTim committed:
Reduce compilation time with withApp

0xTim committed:
Migrate the final test case over to Swift Testing

0xTim committed:
Remove XCTVapor

0xTim committed:
Tidy up old testing stuff

0xTim committed:
Tidy up old testing stuff

0xTim committed:
Start process of removing the content configuration singleton

0xTim committed:
More tidy ups

0xTim committed:
Start moving stuff over from global content configuration

0xTim committed:
Move more stuff away from global content configuration

0xTim committed:
Move ByteBufferAllocator out of Core

0xTim committed:
More global removes

0xTim committed:
Move validations away from global content configuration

0xTim committed:
Goodbye singleton

0xTim committed:
Fix the tests

0xTim committed:
Move Metrics tests back since it didn't help and disable until we're fully async and test traits work

0xTim committed:
Move the ViewRenderer over to use normal dependency injection

0xTim committed:
Get HTTPChannelHandler compiling

0xTim committed:
Fix compilation

0xTim committed:
Switch over to very hacked version of the new server

0xTim committed:
Fancy logging

0xTim committed:
Migrate to HTTPTypes Method away from NIO

0xTim committed:
Start converting HTTPHeaders and HTTPResponseStatus

0xTim committed:
Start sorting out header names

0xTim committed:
More migrations

0xTim committed:
Pull in new multipartkit

0xTim committed:
Some fixes

0xTim committed:
Get Vapor core migrated over to HTTPFields

0xTim committed:
Finish Migrating to HTTPTypes

0xTim committed:
Start migrating the responder over

0xTim committed:
Switch Vapor responder to async

0xTim committed:
Remove AsyncResponseEncodable

0xTim committed:
Migrate the tests to async/await

0xTim committed:
Fix cors test

0xTim committed:
Merge branch 'test-traits-metrics' into vapor-5

0xTim committed:
Pass through address when starting new server

0xTim committed:
Stop tests hanging and handle some port 0 test cases

0xTim committed:
Get some more tests passong

0xTim committed:
Adopt existential any

0xTim committed:
Adopt existential any to the tests

0xTim committed:
Stop the tracing tests from hanging

0xTim committed:
Correcly hook up request's URI

0xTim committed:
Stop more test hangs

0xTim committed:
Add failing test for request body

0xTim committed:
Hack in body support

0xTim committed:
Start tidying up the tests

0xTim committed:
Get some tests passing, disabling for now where necessary

0xTim committed:
Introduce server config to allow hostname/port to be set as before

0xTim committed:
Start working on the requset tests

0xTim committed:
Move tests to new server config

0xTim committed:
Typo

0xTim committed:
Get tracing middleware working

0xTim committed:
Tidy up tracing now we're fully future based

0xTim committed:
Tidy up server start

0xTim committed:
Disable the tests that we know won't work at the moment

0xTim committed:
All tests that should work currently running without hanging

0xTim committed:
Wait for the port properly without needing to sleep. Disable hanging tests

0xTim committed:
Enable report metrics flag

0xTim committed:
Fix typo

0xTim committed:
Fix cookie parsing

0xTim committed:
Fix up some header tests

0xTim committed:
Header tests working

0xTim committed:
Tidy up BindAddress

0xTim committed:
Clean up code

0xTim committed:
Remove redundant warning

0xTim committed:
Lots of tidy ups

0xTim committed:
More tidy ups

0xTim committed:
Fix some Sendable stuff

ptoffy committed:
Update `File` multipart conformance

0xTim committed:
Bring back most of the client tests

0xTim committed:
Tidy up existing tests

0xTim committed:
Migrate testing application over to new helper

0xTim committed:
Start exposing the new body and get all client tests working

0xTim committed:
Don't allow chaining on .test() calls

0xTim committed:
Rough out new testing API for multiple requests in a test

0xTim committed:
Test bug annotations

0xTim committed:
Get the new API working

0xTim committed:
Fix up last route test with new API

0xTim committed:
Merge branch 'main' into vapor-5

0xTim committed:
Fix FileMiddleware for Vapor 5

0xTim committed:
Migrate the new tests for Vapor 5

0xTim committed:
Bump to 6.1

0xTim committed:
Merge branch 'main' into vapor-5

gwynne committed:
A big old pile of fixing up references in comments, removing spurious uses of `return`, cleaning up code style, adding missing uses of self., removing unneeded imports, and rewriting <E:Encodable>(_: E) as (some Encodable).

gwynne committed:
AuthenticationCache no longer needs UnsafeAuthenticationBox.

gwynne committed:
Adjust BasicCodingKey and its call sites for the fact that CodingKeyRepresentable can now be relied on.

gwynne committed:
Fix missing Sendable requirement on JSONEn/Decoder userInfo.

gwynne committed:
Use String(validating:as:) instead of using unsafe pointers.

gwynne committed:
Instead of silencing all Concurrency warnings from RoutingKit, mark the TrieRouter in DefaultResponder as nonisolated(unsafe) (we use it safely in practice)

gwynne committed:
Use Clock (via a trick borrowed from SwiftTesting) instead of Dispatch to get the uptime in nanoseconds

gwynne committed:
Remove the weird random retroactive conformance of UUID to LosslessStringConvertible.

gwynne committed:
URI and URI.Scheme should be ExpressibleByStringLiteral, not ExpressibleByStringInterpolation.

0xTim committed:
Reminder

0xTim committed:
Update NIO dependency and remove our equatable implementation

0xTim committed:
Update gitignore

gwynne committed:
Fix recording of elapsed time in metrics

gwynne committed:
Junk a bunch of stuff in URI that compensates for cross-platform differences that no longer exist in Swift 6, and make it ExpressibleByStringInterpolation again

gwynne committed:
Work around a Sendable issue with JSONEncoder's userInfo on Linux

0xTim committed:
Merge branch 'main' into vapor-5

0xTim committed:
no point testing parallel if we're default anyway

0xTim committed:
Turn on Android testing

Jinxiansen commented:
Hi @0xTim @gwynne 
May I know any plans for Vapor 5 release, are there any blockers?
We have some Vapor community groups in China that are really looking forward to this.
Looking forward to your reply, thanks!

0xTim commented:
@Jinxiansen we're waiting to see how macros play out but ticking through the evolution. I'm hoping we should have an alpha soon that we can use to start refining the API

0xTim committed:
Merge branch 'main' into vapor-5

MojtabaHs committed:
test: fix `system` in internal classes (#3339)

MojtabaHs committed:
fix! typo in `unexpected` - (Breaking change) (#3337)

* docs: fix `unexpected` results

* fix!: add the corrected wording for the error case

This change does not affect the library but it prevents any old switch case on the error.

* refactor: use the new corrected case

* feat: elevate the deprecated signature access level to public

MojtabaHs committed:
test: fix `configuration` argument (#3338)

0xTim committed:
Merge branch 'main' into vapor-5

0xTim committed:
Fix compilation error

0xTim committed:
Adopt service lifecycle

0xTim committed:
Remove the service type

0xTim committed:
Shutdown the thread pool correctly

0xTim committed:
Remove the thread pool in favour of the singleton

0xTim committed:
Simplyfy the password stuff and make it async by default

0xTim committed:
Migrate PasswordHasher to dependency injection

0xTim committed:
Fix the password tests

0xTim committed:
Stanardise the service injection

0xTim committed:
Migrate the cache to new DI

0xTim committed:
Fix the tests

0xTim committed:
Improve the tests

0xTim committed:
Migrate the responder over to new DI

0xTim committed:
Migrate endpoint cache tests over

0xTim committed:
Migrate routes away from storage

0xTim committed:
Migrate Client over to use DI

0xTim committed:
Start migrating the tests

0xTim committed:
Request tests passing

0xTim committed:
Migrate the logger

0xTim committed:
Get the tests passing

0xTim committed:
Tidy up the tests

0xTim committed:
Migrate away using future based AHC calls

0xTim committed:
Sendable Metadata fixes

0xTim committed:
Remove the old HTTP server

0xTim committed:
More tidy up from the old HTTP server

0xTim committed:
Remove AsyncKit since we don't use it

0xTim committed:
Move bcrypt behind a package trait

0xTim committed:
Merge branch 'main' into vapor-5

ptoffy committed:
Use new router

0xTim committed:
Fix compiler errors

0xTim committed:
Fix merge issue

0xTim committed:
Migrate to swift testing

0xTim committed:
[Vapor 5] Migrate AHC, bcrypt, websockets and TLS to behind a package trait (#3356)

* Move websockets behind a trait

* Don't pull in TLS stuff unless behind a trait

* Move client behind a trait

* Add CI step to test traits disabled

* Fix typo

* Use better name for websocket trait

* Fix Sendable issue

* D'oh

* Update test.yml

* Make it compile

* Fix crash when running tests with package traits

* Fix traits workflows

---------

Co-authored-by: Gwynne Raskind <gwynne@vapor.codes>

0xTim committed:
Fix WebSockets trait (#3367)

* Fix websocket trait

* Fix Android

ajevans99 committed:
Add dynamic origin strategy & correct Vary: Origin handling in CORS middleware (#3340)

0xTim committed:
Add Multipart Trait (#3368)

* Add multipart trait

* Move multipart code to behind a trait

0xTim committed:
Depend on Crypto 4 (#3373)

Pull in Crypto 4

0xTim committed:
Remove EventLoops from Vapor's API (#3366)

0xTim committed:
Merge branch 'main' into vapor-5

0xTim committed:
Tidy up comment

ptoffy committed:
Allow unicode routes (#3400)

0xTim committed:
Enable MemberImportVisibility and NonIsolatedNonSending (#3409)

0xTim committed:
Merge branch 'main' into vapor-5

0xTim committed:
Fix Sendable issue

0xTim committed:
Fix package traits

0xTim committed:
Disable provider tests for now

0xTim committed:
Disable Android

0xTim committed:
Async RouteCollection (#3410)

Make 's  async to allow things like file reading when setting up controllers

ptoffy committed:
Add .all routes builder method for catching all HTTP methods (#3401)

* Add `.all` routes builder method for all HTTP methods

* Add head, options and a test

* Update Tests/VaporTests/RouteTests.swift

Co-authored-by: Tim Condon <0xTim@users.noreply.github.com>

* Add DocC

---------

Co-authored-by: Tim Condon <0xTim@users.noreply.github.com>

fpseverino committed:
Remove ConsoleKit and prepare for Swift Configuration (#3403)

ptoffy committed:
Update tests to behave correctly without AHC (#3418)

* Update tests to behave correctly without AHC

* Use HTTPClient.shared

* Update CI

fpseverino committed:
Adopt Swift Configuration (#3421)

0xTim committed:
Merge branch 'main' into vapor-5

0xTim committed:
Fix the metrics tests

0xTim committed:
Fix up middleware tests

0xTim committed:
Tidy up tests

0xTim committed:
Fix authentication tests

0xTim committed:
Update tests for HTTP Types syntax

0xTim committed:
Gitignore update

0xTim committed:
Start bringing back ConditionalCompressionTests

0xTim committed:
Fix CI

0xTim committed:
More fixes

NeedleInAJayStack committed:
Adds OTel metrics middleware (#3324)

* feat: Adds OTel metrics middleware

This adds a middleware that implements the OTel standard http metrics.

# Conflicts:
#	Package.swift
#	Tests/VaporTests/MiddlewareTests.swift

# Conflicts:
#	Package.swift

* fix: Fixes rebasing conflicts

In particular:
1. MetricsMiddleware changed from DispatchTime to ContinuousClock
2. Tests converted from XCTest to Testing

* fix: Detects Abort status codes

* refactor!: Removes Default Resolver metrics

* test: Adopts TaskLocalMetricsSystem

0xTim committed:
Macro Routing (#3394)

0xTim committed:
Merge branch 'main' into vapor-5

ptoffy committed:
Add stored contentType to File (#3450)

Add contentType to File

0xTim committed:
Merge branch 'main' into vapor-5

0xTim committed:
Fix merge errors

kylebrowning committed:
Add @AuthMiddleware macro for type-safe authentication (#3447)

* Add @AuthMiddleware macro for type-safe authentication

Introduces @AuthMiddleware(Type.self, middleware...) that works with
HTTP method macros to extract authenticated users via req.auth.require()
and wrap route registration with .grouped() middleware in @Controller.

* Fix the Authentication type macro

* Add some integration tests

* Fix traits build

---------

Co-authored-by: Tim Condon <0xTim@users.noreply.github.com>
Co-authored-by: Tim <0xtimc@gmail.com>

0xTim committed:
Merge branch 'main' into vapor-5

0xTim committed:
Adopt New HTTP Servr (#3465)

* Add dependency on HTTP Server library

* Repo is now public

* Update to new target

* Bump the platforms to support the new server

* Start removing old new sever

* Start mashing in the new body stuff

* Add a body writer

* Start introducing handler

* Conform it to Vapor's app

* Fix ambiguous configuration type

* Fix the client tests

* Fix path conversion

* Fix percent encoding of paths

* Stop test crash

* Narrow down application test hangs

* Fix hanging application test

* Tidy up

* Start migrating to structured concurrency

* Test update

* Fix a rebase issue

* Update for new changes

* Update flags to fix linker error

* Disable the failing tests

* tidy up

* Add graceful shutdown test

* 6.4 CI
